lab-michaelm/test/test-router-member.js

```diff
@@ -0,0 +1,208 @@
+
+// test env
+require('dotenv').config({path: `${__dirname}/../.test.env`});
+
+// npm mods
+const expect = require('expect');
+const superagent = require('superagent');
+// app mods
+const server = require('../lib/server.js');
+const clearDB = require('./lib/clear-db.js');
+const Leader = require('../model/leader.js');
+const mockMember = require('./model/mock-member.js');
+const mockLeader = require('./model/mock-leader.js');
+
+const API_URL = process.env.API_URL;
+
+describe('--------Testing member routes----------', () => {
+  before(server.start);
+  after(server.stop);
+  afterEach(clearDB);
+
+  describe('Testing POST /api/member', () => {
+    it('Should return a new member', () => {
+      let tempMember, tempLeader;
+      return mockLeader.createOne()
+      .then(leader => {
+        tempLeader = leader;
+        return superagent.post(`${API_URL}/api/member`)
+        .send({
+          firstName: 'Mike',
+          lastName: 'Miller',
+          availabilityDate: ['07/02/2017', '07/09/2017'],
+          userName: 'bling',
+          leader: [leader._id.toString()],
+        });
+      })
+      .then(res => {
+        expect(res.status).toEqual(200);
+        expect(res.body.firstName).toEqual('Mike');
+        expect(res.body.lastName).toEqual('Miller');
+        expect(res.body.availabilityDate).toEqual(['07/02/2017', '07/09/2017']);
+        expect(res.body.leader).toEqual(tempLeader._id.toString());
+        expect(res.body.userName).toEqual('bling');
+        expect(res.body._id).toExist();
+        tempMember = res.body;
+        return Leader.findById(tempLeader._id);
+      })
+      .then(leader => {
+        expect(leader.member.length).toEqual(1);
+        expect(leader.member[0].toString()).toEqual(tempMember._id.toString());
+      });
+    });
+    it('Should respond with a 400 status code', () => {
+      return superagent.post(`${API_URL}/api/member`)
+      .send({
+        firstName: 'Mike',
+        lastName: 'Miller',
+        availabilityDate: ['07/02/2017', '07/09/2017'],
+        userName: 'bling',
+        leader: '654684653548',
+      })
+      .then(res => {
+        throw res;
+      })
+      .catch(res => {
+        expect(res.status).toEqual(400);
+      });
+    });
+    it('Should respond with a 409 status code', () => {
+      let tempLeader;
+      return mockLeader.createOne()
+      .then(leader => {
+        tempLeader = leader;
+        return superagent.post(`${API_URL}/api/member`)
+        .send({
+          firstName: 'Mike',
+          lastName: 'Miller',
+          availabilityDate: ['07/02/2017', '07/09/2017'],
+          userName: 'bling',
+          leader: leader._id.toString(),
+        });
+      })
+      .then(() => {
+        return superagent.post(`${API_URL}/api/member`)
+      .send({
+        firstName: 'Mike',
+        lastName: 'Miller',
+        availabilityDate: ['07/02/2017', '07/09/2017'],
+        userName: 'bling',
+        leader: tempLeader._id.toString(),
```

Comment:
Seems like we are making a post request to test a post request. This is bad practice. No test should require other requests being sent. I believe we are able to mock a member at this point and send the same mocked member on the request.